cmd/geth/version_check.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2020 The go-ethereum Authors
+// This file is part of go-ethereum.
+//
+// go-ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// go-ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with go-ethereum. If not, see <http://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/jedisct1/go-minisign"
+	"gopkg.in/urfave/cli.v1"
+)
+
+var gethPubKeys []string = []string{
+	//@holiman, minisign public key FB1D084D39BAEC24
+	"RWQk7Lo5TQgd+wxBNZM+Zoy+7UhhMHaWKzqoes9tvSbFLJYZhNTbrIjx",
+	//minisign public key 138B1CA303E51687
+	"RWSHFuUDoxyLEzjszuWZI1xStS66QTyXFFZG18uDfO26CuCsbckX1e9J",
+	//minisign public key FD9813B2D2098484
+	"RWSEhAnSshOY/b+GmaiDkObbCWefsAoavjoLcPjBo1xn71yuOH5I+Lts",
+}
+
+type vulnJson struct {
+	Name        string
+	Uid         string
+	Summary     string
+	Description string
+	Links       []string
+	Introduced  string
+	Fixed       string
+	Published   string
+	Severity    string
+	Check       string
+	CVE         string
+}
+
+func versionCheck(ctx *cli.Context) error {
```

Comment:
for what is this used for?

Reply:
it is a flag:
The version-check command fetches vulnerability-information from https://geth.ethereum.org/docs/vulnerabilities/vulnerabilities.json, 
and displays information about any security vulnerabilities that affect the currently executing version.

Reply:
nice